src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ActionRow, Card, Dropdown, Icon, IconButton, useToggle,
+} from '@openedx/paragon';
+import { EditOutline as EditIcon, MoreVert as MoveVertIcon } from '@openedx/paragon/icons';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { scrollToElement } from '../../course-outline/utils';
+import messages from './messages';
+
+const CourseXBlock = ({
+  id, title, unitXBlockActions, shouldScroll, ...props
+}) => {
+  const courseXBlockElementRef = useRef(null);
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const intl = useIntl();
+
+  const onXBlockDelete = () => {
+    unitXBlockActions.handleDelete(id);
+    closeDeleteModal();
+  };
+
+  useEffect(() => {
+    // if this item has been newly added, scroll to it.
+    if (courseXBlockElementRef.current && shouldScroll) {
+      scrollToElement(courseXBlockElementRef.current);
+    }
+  }, []);
+
+  return (
+    <div ref={courseXBlockElementRef} {...props}>
+      <Card className="mb-1">
+        <Card.Header
```

Comment:
Missing drag handle. Is that in scope for this ticket?

Reply:
No, the Drag&Drop functionality will be done in next iterations.